wagtail/models/__init__.py

```diff
@@ -3003,26 +3103,30 @@ def editable_pages(self):
 
         editable_pages = Page.objects.none()
 
-        for perm in self.permissions.filter(permission_type="add"):
-            # user has edit permission on any subpage of perm.page
-            # (including perm.page itself) that is owned by them
-            editable_pages |= Page.objects.descendant_of(
-                perm.page, inclusive=True
-            ).filter(owner=self.user)
+        for page in self.get_pages_for_perms(["add"]):
+            # user has edit permission on any subpage of page
+            # (including page itself) that is owned by them
+            editable_pages |= Page.objects.descendant_of(page, inclusive=True).filter(
+                owner=self.user
+            )
 
-        for perm in self.permissions.filter(permission_type="edit"):
-            # user has edit permission on any subpage of perm.page
-            # (including perm.page itself) regardless of owner
-            editable_pages |= Page.objects.descendant_of(perm.page, inclusive=True)
+        for page in self.get_pages_for_perms(["edit"]):
+            # user has edit permission on any subpage of page
+            # (including page itself) regardless of owner
+            editable_pages |= Page.objects.descendant_of(page, inclusive=True)
 
         return editable_pages
 
-    def can_edit_pages(self):
-        """Return True if the user has permission to edit any pages"""
+    @cached_property
+    def _can_edit_pages(self):
+        """Return True if the user has permission to edit any pages."""
         return self.editable_pages().exists()
 
+    def can_edit_pages(self):
+        return self._can_edit_pages
```

Comment:
Issue: This seems like an anti-pattern. Why not just name the above cached property `can_edit_pages` and avoid the need for a separate method

Reply:
Because the other parts of the code expect `can_edit_pages` to be a callable:
```python
UserPagePermissionProxy(request.user).can_edit_pages()
```